.github/workflows/go.yaml

```diff
@@ -2,76 +2,136 @@ name: Go
+    - uses: actions/checkout@v2
+    - name: Install python3 deps
+      run: pip3 install requests
+#   TODO: Consider cache removal. Caching means that jobs and periodis won't be able to verify that external depepndencies are still present and working.
+    - name: Cache tools
+      id: cache-tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ format('{0}/bin', env.GOPATH) }}
+        key: ${{ runner.os }}-${{ hashFiles('./install-dependencies.sh') }}
+    - name: Install deps
+      if: ${{ steps['cache-tools'].outputs['cache-hit'] != 'true' }}
+      run: ./install-dependencies.sh
+    - name: Test integration
+      run: make test-integration --warn-undefined-variables
 
-      - name: Unit Tests
-        working-directory: ${{ env.working-directory }}
-        run: hack/run_and_diff.py -v make test
+  images:
+    name: Build images
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./Dockerfile
+        platforms: linux/amd64
+        push: false
 
-      - name: Build
-        working-directory: ${{ env.working-directory }}
-        run: make local-build
+  # TODO: Add e2e - use the same image sha from images step
 
-      - name: Login to DockerHub
-        if: ${{ github.event_name != 'pull_request' }}
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+  # TODO: Add upgrade-e2e - use the same image sha from images step
 
-      - name: Build nightly image
-        if: ${{ github.event_name != 'pull_request' }}
-        working-directory: ${{ env.working-directory }}
-        run: make nightly
+  # Dummy step for different propromotion jobs to depend on
+  success:
+    name: All tests successfull
+    runs-on: ubuntu-20.04
+    needs:
+    - verify
+    - build-and-test
+    - test-integration
+    - images
+    # TODO: Depend on e2e when available
+    # TODO: Depend on upgrade-e2e when available
+    steps:
+    - run: /bin/true
 
-      - name: Publish nightly image
-        if: ${{ github.event_name != 'pull_request' }}
-        working-directory: ${{ env.working-directory }}
-        run: make docker-push
+  promote-on-merge:
+    name: Promote artifacts on merge
+    runs-on: ubuntu-20.04
+    needs: [success]
+    if: github.event_name != 'pull_request' && github.ref == 'refs/heads/master'
+    steps:
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    # TODO: replace with pushing to latest
+    # TODO: use the same image sha from images step - has to match what we used for e2e
+    - name: Build nightly image
+      run: make nightly
+    - name: Publish nightly image
+      run: goreleaser --snapshot --rm-dist --config=.goreleaser-nightly.yml
```

Comment:
This doesn't publish image, it only builds it. You have to explicitly use docker push

Reply:
fixed, thx